Lib/gftools/packager.py

```diff
@@ -671,7 +674,10 @@ def _upstream_conf_from_yaml_metadata(
                                        , allow_flow_style=True)
   for k,v in upstream_conf.items():
     if v is None: continue
-    upstream_conf_yaml[k] = v
+    if k == "category":
+      upstream_conf_yaml[k] = v._values
```

Comment:
We cannot simply do `upstream_conf_yaml[k] = v` for the category since it's not a list.

Reply:
I can't actually reproduce the issue, for me it works when `category` is a `<class 'google.protobuf.pyext._message.RepeatedScalarContainer'>`.

Maybe this is an issue with the version of the `protobuf` package?

Despite of that, maybe it would be nicer to do this before in line 654:

```py
   upstream_conf.update({
      'designer': metadata.designer or None,
      'category': list(metadata.category) or None,
      'name': metadata.name  or None,
      # we won't get this just now in most cases!
      'repository_url': metadata.source.repository_url or None,
    })
```

Reply:
I'm using protobuf 3.19.1 which seems to be the latest. I'm able to replicate it. I'll use your last proposal since that is definitely the cleanest. Thank you.